support-services/src/main/scala/com/gu/support/redemption/CodeValidationResult.scala

```diff
@@ -0,0 +1,30 @@
+package com.gu.support.redemption
+
+import com.gu.support.redemption.corporate.CorporateCodeValidator.CorporateId
+
+sealed abstract class CodeValidationResult(val clientCode: String)
+
+sealed abstract class ValidCode(clientCode: String) extends CodeValidationResult(clientCode)
+
+sealed abstract class InvalidCode(clientCode: String) extends CodeValidationResult(clientCode)
+
+case object CodeNotFound extends CodeValidationResult("code_not_found")
+
+case object CodeMalformed extends InvalidCode("code_malformed")
+
+case object CodeAlreadyUsed extends InvalidCode("code_already_used")
+
+case object CodeExpired extends InvalidCode("code_expired")
+
+case object InvalidReaderType extends InvalidCode("invalid_reader_type")
+
+object ValidGiftCode { val clientCode = "valid_gift_code"}
+
+case class ValidGiftCode(subscriptionId: String) extends ValidCode(ValidGiftCode.clientCode)
```

Comment:
regarding ValidGiftCode and ValidCode, assuming the distinction is necessary, would be nice to capture that distinction in the name

Reply:
`ValidCode` is an abstract class, it's extended by `ValidGiftCode` and `ValidCorporateCode` to represent the set of valid code types.

Reply:
ok got it now! 👍